docs/contributing/contributing-rules.md

```diff
@@ -1,102 +1,334 @@
-    - `best-practice`
-    - `correctness`
-    - `maintainability`
-    - `performance`
-- Include the `technology` field. This is usually the library or framework the rule is targeting, for example `django`. If it's for the language itself, use only the language name, for example `python`.
-- Include the `references` field. References can provide additional context to users of the rule. It is good practice to include at least one reference for each rule.
-
-Rule examples:
-- [python.lang.security.deserialization.avoid-pyyaml-load.yaml](https://semgrep.dev/orgs/-/editor/r/python.lang.security.deserialization.avoid-pyyaml-load.avoid-pyyaml-load)
-- [python/flask/security/dangerous-template-string.yaml](https://semgrep.dev/orgs/-/editor/r/python/flask/security/dangerous-template-string.yaml)
-- [python/flask/security/audit/render-template-string.yaml](https://semgrep.dev/orgs/-/editor/r/python/flask/security/audit/render-template-string.yaml)
-- [javascript/lang/best-practice/assigned-undefined.yaml](https://semgrep.dev/orgs/-/editor/r/javascript/lang/best-practice/assigned-undefined.yaml)
-- [java/rmi/security/server-dangerous-class-deserizaliation.yaml](https://semgrep.dev/orgs/-/editor/r/javascript/lang/best-practice/assigned-undefined.yaml)
+MEDIUM confidence rules can use Semgrep advanced features such as `metavariable-comparison` or `taint mode`, but with some false positives. See examples below:
+
+- https://semgrep.dev/playground/r/javascript.express.security.audit.express-ssrf.express-ssrf
+- https://semgrep.dev/playground/r/javascript.express.security.express-xml2json-xxe.express-xml2json-xxe?editorMode=advanced
+
+```
+confidence: MEDIUM 
+```
+
+##### LOW
+
+Low confidence rules generally find something which appears to be dangerous while reporting a lot of false positives. See examples below:
+
+- https://semgrep.dev/playground/r/php.lang.security.eval-use.eval-use
+- https://semgrep.dev/playground/r/javascript.browser.security.dom-based-xss.dom-based-xss?editorMode=advanced
+
+```
+confidence: LOW 
+```
+
+#### Likelihood
+
+Specify how likely it is that an attacker can exploit the issue that has been found. The possible values are following:
+
+##### HIGH
+
+HIGH likelihood rules tend to be something which is easily reasoned with if found very likely to be a concern:
+
+Some general examples include:
+- The use of weak encryption
+  - https://semgrep.dev/playground/r/go.lang.security.audit.crypto.use_of_weak_rsa_key.use-of-weak-rsa-key?editorMode=advanced
+- Disabled security feature in a configuration
```

Comment:
@LewisArdern no rule attached to this list item.

Reply:
I dont think we have any good 'public rules' which highlight this.